Generator/Generator.cs

```diff
@@ -1,25 +1,37 @@
-﻿using System.Linq;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
 using Repository;
+using Generator.Services.Writer;
+
 using StrongInject;
 
 namespace Generator
 {
     public static class Generator
     {
-        public static void Write(string[] projects)
+        public static void Write(IEnumerable<string> projects, string outputDir = "output")
         {
             var repository = new Repository.Repository();
             var loadedProjects = repository.Load(FileResolver.ResolveFile, projects).ToList();
 
             Log.Information("Ready to write.");
 
-            var writerService = new Container().Resolve().Value;
-            foreach (LoadedProject proj in loadedProjects)
+            try
```

Comment:
Please don't add exception handling here. This is the task of the caller. Even if it is the build project.

After merging this branch into develop  we should probably remove the build project and  enhance the build pipeline to use the same tool which 3rd party devs would use. This would require to add the different parameters of the build project to the tool which itself then uses the generator.

This means that the top level caller can decide how to handle exceptions and not the generator in between as all non critical exceptions are already handled inside the generator (like detecting var args arguments for example).